### DISCUSSION TITLE
Question: why is `_errors` an array?

### DISCUSSION THREAD

**Discussion by nikitazdvijkov** [colinhacks/zod#1866] [Q&A] [answered]
Why are the error messages always in array form? I have not found an example in the docs of multiple error messages for one field. I cannot figure out how to make the error messages array have more than one element.

Example:

```js
x.error?.flatten()
```

Gives:

```js
{
  formErrors: [],
  fieldErrors: { 
    username: [ 'Expected number, received string' ],
    password: [ 'Must contain capital letters' ]
  }
}
```

Why does the `username` key correspond to an array? That array is supposed to be an array of strings (`string[]`), right? Why is it an array if it's always only one element? **Why not just have the `username` key correspond to a string?**

This isn't just with the `.flatten()` method. It's the way errors are represented everywhere.

**Comment by JacobWeisenburger** [comment, marked as answer]
Errors are an array because sometimes there is more than 1 thing wrong. And it's easier to know it's always an array than it is to try to deal with it being a `string` when there is 1 thing wrong and a `string[]` many things are wrong.

```ts
const schema = z.object( {
    foo: z.string(),
    bar: z.object( {
        a: z.string(),
        b: z.number(),
    } ),
} )

const result = schema.safeParse( {
    foo: 42, // Expected string, received number
    bar: {
        a: 1, // Expected string, received number
        b: 'b', // Expected number, received string
    }
} )

!result.success && console.log( result.error.flatten() )
// {
//     formErrors: [],
//     fieldErrors: {
//       foo: [ "Expected string, received number" ],
//       bar: [ "Expected string, received number", "Expected number, received string" ]
//     }
// }
```

**Reply by nikitazdvijkov** [reply]
Thanks for the example!
